backend/cmmgen_state.ml

```diff
@@ -95,13 +113,38 @@ let compare_structured_constants c1 c2 =
     if c <> 0 then c else compare_constant_lists l1 l2
   | Const_float_array l1, Const_float_array l2 -> compare_float_lists l1 l2
   | Const_string s1, Const_string s2 -> String.compare s1 s2
-  | Const_vec128 { high = l0; low = l1 }, Const_vec128 { high = r0; low = r1 }
-    ->
-    let cmp = Int64.compare l0 r0 in
-    if cmp = 0 then Int64.compare l1 r1 else cmp
+  | ( Const_vec128 { word0 = l0; word1 = l1 },
+      Const_vec128 { word0 = r0; word1 = r1 } ) ->
+    List.compare Int64.compare [l0; l1] [r0; r1]
```

Comment:
(wouldn't that be a tiny bit better
with an array? Not that it really
matters...)

Reply:
It would, but `Array.compare` is not provided by the stdlib >_>

Reply:
But it is in `Misc.Stdlib.Array`.